dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -418,6 +414,60 @@ private void searchAndPopulate(final String luceneQuery, final User user,
         }
     }
 
+    private void checkSiteLanguageVanities(final String siteId,
+                                           final Long languageId,
+                                           final Boolean includedSystemHostOnLuceneQuery) {
+
+        if (null == this.vanityUrlServices.getCachedVanityUrlList(siteId, languageId)) {
+
+            this.vanityUrlServices.setCachedVanityUrlList(siteId, languageId, Collections.EMPTY_LIST);
+        }
+
+        if (includedSystemHostOnLuceneQuery && !Host.SYSTEM_HOST.equals(siteId) &&
+                null == this.vanityUrlServices.getCachedVanityUrlList(Host.SYSTEM_HOST, languageId)) {
+
+            this.vanityUrlServices.setCachedVanityUrlList(Host.SYSTEM_HOST, languageId, Collections.EMPTY_LIST);
+        }
+    } // checkSiteLanguageVanities.
+
+    private void setEmptyCaches(final String siteId,
+                                final Long languageId,
+                                final Boolean includedSystemHostOnLuceneQuery) {
+
+        if (null != siteId && null != languageId) {
+            this.vanityUrlServices.setCachedVanityUrlList(
+                    siteId, languageId,
+                    Collections.EMPTY_LIST);
+
+            if (includedSystemHostOnLuceneQuery && !Host.SYSTEM_HOST.equals(siteId)) {
+                this.vanityUrlServices.setCachedVanityUrlList(
+                        Host.SYSTEM_HOST, languageId,
+                        Collections.EMPTY_LIST);
+            }
+        }
+    } // setEmptyCaches.
+
+    private void addSecondaryVanityURLCacheCollection(final List<VanityUrl> vanityUrls) {
+
+        final Map<SiteLanguageKey, ImmutableList.Builder<CachedVanityUrl>> vanityPerSiteLanguageMap
+                = map();
+        SiteLanguageKey key = null;
+
+        for (final VanityUrl vanityUrl : vanityUrls) {
+
+            key = new SiteLanguageKey(vanityUrl.getSite(), vanityUrl.getLanguageId());
+            if (!vanityPerSiteLanguageMap.containsKey(key)) {
+
+                vanityPerSiteLanguageMap.put(key, new ImmutableList.Builder());
+            }
+
+            vanityPerSiteLanguageMap.get(key).add(new CachedVanityUrl(vanityUrl));
+        }
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'k', 'vanityUrlBuilder' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)